webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/AspectDeploymentProcessor.java

```diff
@@ -68,6 +78,8 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
             WSLogger.ROOT_LOGGER.tracef("%s start: %s", aspect, unit.getName());
             ClassLoader origClassLoader = WildFlySecurityManager.getCurrentContextClassLoaderPrivileged();
             try {
+                dep.addAttachment(Boolean.class, isHttpsOnly(phaseContext, unit));
```

Comment:
@jimma Is 'Boolean' an acceptable type for this? 

Well, you'll decide whether the JBoss WS fix gets merged and whatever is used there is what will be needed here. :)

Reply:
If not, this could be replace with a Map<Integer, String> to map port to protocol.